parsers/NI.py

```diff
@@ -0,0 +1,347 @@
+
+    :return: tuple(production, datetime_datetime)
+    """
+
+    type_translator = {
+        'EOLICO': 'wind',
+        'GEOTERMICO': 'geothermal',
+        'BIOMASA': 'biomass',
+        'HIDROELECTRICO': 'hydro',
+        'SOLAR': 'solar',
+
+        # all "thermal" / fossil fuel is oil - see comment at PLANT_CLASSIFICATIONS
+        'TERMICO BUNKER': 'oil'
+    }
+
+    response = requests_obj.get(SUMMARY_URL)
+    response.encoding = 'utf-8'
+    gentype_html = response.text
+
+    datetime_text = extract_text(gentype_html, 'Consultado a las ', '\'')
+    hour = extract_text(datetime_text, '', ' horas')
+    d = extract_text(datetime_text, 'del dia ')
+    datetime_arrow = arrow.get(d + ' ' + hour, 'DD/MM/YYYY HH')
+    datetime_datetime = arrow.get(datetime_arrow.datetime, TIMEZONE).datetime
+
+    gen_type_text = extract_text(gentype_html, 'Tipo de Generación', 'center:')
+    gen_type_text = extract_text(gen_type_text, '[')
+
+    production = {}
+
+    # One of the generation types is featured by default in the pie chart.
+    # This makes its HTML/JS specification be different.
+    # The type which is featured also differs from hour to hour.
+    featured_type = extract_text(gen_type_text, '{', '}')
+    featured_type_props = featured_type.split(',')
+    featured_type_dict = {x.split(':')[0].strip(): x.split(':')[1].strip()
+                          for x in featured_type_props}
+    featured_type_name = featured_type_dict['name'].replace('\'', '')
+    featured_type_val = float(featured_type_dict['y'])
+
+    featured_type_standard_name = type_translator[featured_type_name]
+    production[featured_type_standard_name] = featured_type_val
+
+    # The remaining, non-featured generation types are all formatted the same.
+    other_types = extract_text(gen_type_text, '}')
+    other_types_props = [t.split(',') for t in other_types.split('[')]
+    for other_type in other_types_props:
+        name = other_type[0]
+        val = other_type[1]
+        if name:
+            standard_name = type_translator[name.replace('\'', '')]
+            production[standard_name] = float(val.replace(']', '').strip())
+
+    return production, datetime_datetime
+
+
+def fetch_production(country_code='NI', session=None):
+    """Requests the last known production mix (in MW) of Nicaragua.
+
+    Arguments:
+    country_code       -- ignored here, only information for NI is returned
+    session (optional) -- request session passed in order to re-use an existing session
+    """
+
+    requests_obj = session or requests.session()
+
+    # We're currently using the summary page (SUMMARY_URL, via get_production_from_summary())
+    # rather than the detailed map page (MAP_URL, via get_production_from_map())
+    # in order to get solar production.
+    production, data_datetime = get_production_from_summary(requests_obj)
+
+    data = {
+        'datetime': data_datetime,
+        'countryCode': country_code,
+        'production': production,
```

Comment:
Yep sure! I wasn't sure how important this was when there was already a mix of a few modes. I'll add it.